components/eamxx/src/share/field/field_alloc_prop.cpp

```diff
@@ -77,6 +77,53 @@ subview (const int idim, const int k, const bool dynamic) const {
   return props;
 }
 
+FieldAllocProp FieldAllocProp::subview(const int idim,
+                                       const int k_beg,
+                                       const int k_end) const {
+  EKAT_REQUIRE_MSG(
+      is_committed(),
+      "Error! Subview requires alloc properties to be committed.\n");
+  EKAT_REQUIRE_MSG(idim < m_layout.rank(),
+                   "Error! Dimension index out of bounds.\n");
+  EKAT_REQUIRE_MSG(k_beg < k_end,
+                   "Error! Slice indices are invalid (non-increasing).\n");
+  EKAT_REQUIRE_MSG(
+      k_beg >= 0 && k_end < m_layout.dim(idim),
+      "Error! Slice index range along the idim dimension is out of bounds.\n");
+
+  // Set new layout basic stuff
+  FieldAllocProp props(m_scalar_type_size);
+  props.m_committed = true;
+  props.m_scalar_type_size = m_scalar_type_size;
+  props.m_layout = m_layout.clone();
+  props.m_layout.reset_dim(idim, k_end - k_beg);
+
+  // Output is contiguous if either
+  //  - this->m_contiguous=true AND idim==0
+  //  - m_layout.dim(i)==1 for all i<idim
+  auto is_one = [](int i) { return i == 1; };
+  // we do not currently have the capability for contiguous multi-slice views
+  props.m_contiguous = false;
+
+  props.m_subview_info =
+      SubviewInfo(idim, k_beg, k_end, m_layout.dim(idim));
+
+  // Figure out strides/packs
+  if (idim == m_layout.rank()) {
+    // We're slicing the possibly padded dim, so everything else is as in the
+    // layout, and there is no packing
```

Comment:
With multislice, we can still have packing. We should maintain the same pack size as the parent view.

Reply:
Actually, packing becomes problematic once we slice, because of memory alignment. So yeah, let's keep pack_size_max=1 for safety.